sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -1797,11 +1797,11 @@ SIMPLE_COMMENT
     ;
 
 BRACKETED_EMPTY_COMMENT
-    : '/**/' -> channel(HIDDEN)
+    : '/*' BRACKETED_EMPTY_COMMENT? '*/' -> channel(HIDDEN)
     ;
 
 BRACKETED_COMMENT
```

Comment:
the hint and comment syntax are so similar and it's too hacky to put one as parser rule and one as lexer rule.

Your fix looks correct, but it's hard to reason about and @gengliangwang has pointed our issues twice.

I think we should make them both parser rules (hint needs to create a plan so can't be lexer rule)

Reply:
+1 with @cloud-fan 
The latest code will still fail
```
/*/*foo*//*bar*/*/
```
since you just match the starting `BRACKETED_EMPTY_COMMENT` once.

Reply:
I think even though to create a plan, we still need to distinguish hint and comment syntax.
@gengliangwang I improve the syntax again, please take a look again.

Reply:
> we still need to distinguish hint and comment syntax

We can distinguish them better if they are both parser rules.

Reply:
If this problem could be solved in g4, things would be even simpler. Let me do some hard work and try.

Reply:
Since hint can appear at any position, it's not possible to create a parser rule for it.